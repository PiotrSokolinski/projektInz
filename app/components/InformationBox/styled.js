import styled from 'styled-components'

import { Colors, Media } from '../../themes'

const getColorForType = props => {
  switch (props.type) {
    case 'error':
      return Colors.red
    case 'warning':
      return Colors.ecstasy
    default:
      return Colors.black
  }
}

const getBackgroundColorForType = props => {
  switch (props.type) {
    case 'error':
      return Colors.errorBackground
    case 'warning':
      return Colors.warningBackground
    default:
      return Colors.transparentBlack
  }
}

export const Container = styled.div`
  align-items: center;
  background-color: ${props => getBackgroundColorForType(props)};
  border-radius: 5px;
  display: flex;
  margin: 20px 0;
  padding: 11px;
  width: ${({ fullWidth }) => (fullWidth ? '100%' : '300px')};

  ${Media.mobile`
    width: 100%;
  `};
`

export const Icon = styled.img`
  margin-right: 8px;
`

export const Text = styled.p`
  color: ${props => getColorForType(props)};
  font-size: 13px;
  letter-spacing: 0.02em;
  line-height: 19px;
`
