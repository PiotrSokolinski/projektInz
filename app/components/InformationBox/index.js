/**
 *
 * InformationBox
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { Container, Icon, Text } from './styled'

import { Icons } from '../../themes'

const InformationBox = ({ children, className, fullWidth, icon, type }) => (
  <Container fullWidth={fullWidth} className={className} type={type}>
    {/* {type !== 'warning' && <Icon src={Icons[icon]} />} */}
    <Text type={type}>{children}</Text>
  </Container>
)

InformationBox.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  fullWidth: PropTypes.bool,
  icon: PropTypes.string,
  type: PropTypes.string,
}

InformationBox.defaultProps = {
  children: null,
  className: '',
  fullWidth: false,
  icon: 'exclamationIcon',
  type: 'error',
}

export default InformationBox
