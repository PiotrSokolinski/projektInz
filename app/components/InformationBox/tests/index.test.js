/**
 *
 * Tests for InformationBox
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import InformationBox from '..'

describe('<InformationBox />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<InformationBox />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<InformationBox type="other" fullWidth />)
    expect(firstChild).toMatchSnapshot()
  })

  it('Should render error and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<InformationBox type="error" />)
    expect(firstChild).toMatchSnapshot()
  })
})
