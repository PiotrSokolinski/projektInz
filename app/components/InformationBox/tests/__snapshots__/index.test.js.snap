// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationBox /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-wfp3xe-0 bjgfMP"
  type="other"
>
  <img
    class="styled__Icon-wfp3xe-1 dzrHpH"
    src="IMAGE_MOCK"
  />
  <p
    class="styled__Text-wfp3xe-2 jMStTs"
    type="other"
  />
</div>
`;

exports[`<InformationBox /> Should render error and match the snapshot 1`] = `
<div
  class="styled__Container-wfp3xe-0 iTEjFd"
  type="error"
>
  <img
    class="styled__Icon-wfp3xe-1 dzrHpH"
    src="IMAGE_MOCK"
  />
  <p
    class="styled__Text-wfp3xe-2 igxWPW"
    type="error"
  />
</div>
`;
